### ISSUE TITLE
Add categorical colorbars for plot and plot3d gallery examples

### ISSUE THREAD

**Issue by michaelgrund** [GenericMappingTools/pygmt#1259]
**Description of the problem**

I recognized that some colors of individual gallery examples produced with 6.2.0rc1 show deviations compared to the versions generated with 6.1 (see below). Any ideas what's the reason for that?

**6.1 versions**
![sphx_glr_points_categorical_001](https://user-images.githubusercontent.com/23025878/117105652-c7881180-ad7e-11eb-8ab7-0d8954d75a20.png)

![sphx_glr_scatter3d_001](https://user-images.githubusercontent.com/23025878/117105815-159d1500-ad7f-11eb-8055-4392552548d6.png)

**6.2.0rc1 versions**

![sphx_glr_points_categorical_001_new](https://user-images.githubusercontent.com/23025878/117105775-01591800-ad7f-11eb-9b89-7297c309ff0c.png)

![sphx_glr_scatter3d_001_new](https://user-images.githubusercontent.com/23025878/117105831-1afa5f80-ad7f-11eb-9946-d7f6306a62a9.png)

**Comment by seisman** [comment]
The generated CPTs are different. I think there were some CPT-related changes in upstream GMT. The one generated by GMT 6.2.0rc1 makes more sense to me.

GMT 6.1.1:
```
$ gmt makecpt -Cinferno -T0/3/1 -F+c
0	50.998/10/94.499	L
1	187/54.5/84.5	L
2	251.5/181/25.002	B
B	black
F	white
N	127.5
```

GMT 6.2.0rc1:
```
$ gmt makecpt -Cinferno -T0/3/1 -F+c
0	32.75/12/73.75	L
1	137.25/34/106	L
2	227.37/89.375/50.25	L
3	249/201.25/50.375	B
N	128
```

**Comment by meghanrjones** [comment]
Yes, the previous CPT range was considered a bug that was fixed in this PR: https://github.com/GenericMappingTools/gmt/pull/4446. I think we should update the `series` settings for `makecpt` accordingly in the examples.

**Comment by seisman** [comment]
> I think we should update the `series` settings for `makecpt` accordingly in the examples.

I agree. I think we also need to add a colorbar to these examples. We didn't do that before, because GMT 6.1.1 has some bugs/limitations when plotting categorical colorbars, which should be fixed in GMT 6.2.0rc1.

**Comment by meghanrjones** [comment]
> > I think we should update the `series` settings for `makecpt` accordingly in the examples.
> 
> I agree. I think we also need to add a colorbar to these examples. We didn't do that before, because GMT 6.1.1 has some bugs/limitations when plotting categorical colorbars, which should be fixed in GMT 6.2.0rc1.

Agreed. Are you interested in updating the examples @michaelgrund? I can submit a PR if you are not interested or do not have time.

**Comment by michaelgrund** [comment]
> 
> 
> > > I think we should update the `series` settings for `makecpt` accordingly in the examples.
> > 
> > 
> > I agree. I think we also need to add a colorbar to these examples. We didn't do that before, because GMT 6.1.1 has some bugs/limitations when plotting categorical colorbars, which should be fixed in GMT 6.2.0rc1.
> 
> Agreed. Are you interested in updating the examples @michaelgrund? I can submit a PR if you are not interested or do not have time.

I will work on that the upcoming weekend and will open a corresponding PR @meghanrjones.
